src/test/scala/edu/ie3/simona/agent/participant/EvcsAgentModelCalculationSpec.scala

```diff
@@ -2004,6 +2009,396 @@ class EvcsAgentModelCalculationSpec
+
+      evService.send(
+        evcsAgent,
+        ProvideEvDataMessage(
+          1800,
+          evService.ref,
+          ArrivingEvs(Seq(ev1800)),
+          Some(2700),
+        ),
+      )
+
+      scheduler.expectMsg(Completion(evcsAgent.toTyped, Some(2700)))
+
+      /* TICK 2700
+       * - ev2700 arrives
+       * - charging with 22 kW
+       */
+      scheduler.send(evcsAgent, Activation(2700))
+
+      val ev2700 = EvModelWrapper(evC.copyWithDeparture(5400))
+
+      evService.send(
+        evcsAgent,
+        ProvideEvDataMessage(
+          2700,
+          evService.ref,
+          ArrivingEvs(Seq(ev2700)),
+          None,
+        ),
+      )
+
+      scheduler.expectMsg(Completion(evcsAgent.toTyped, None))
+
+      // TICK 3600: ev900 leaves
+      evService.send(
+        evcsAgent,
+        DepartingEvsRequest(3600, Seq(ev900.uuid)),
+      )
+
+      evService.expectMsgType[DepartingEvsResponse] match {
+        case DepartingEvsResponse(evcs, evModels) =>
+          evcs shouldBe evcsInputModel.getUuid
+          evModels should have size 1
+          evModels.headOption match {
+            case Some(evModel) =>
+              evModel.uuid shouldBe ev900.uuid
+              evModel.storedEnergy should approximate(KilowattHours(8.25))
+            case None => fail("Expected to get at least one ev.")
+          }
+      }
+
+      // TICK 4500: ev1800 leaves
+
+      evService.send(
+        evcsAgent,
+        DepartingEvsRequest(4500, Seq(ev1800.uuid)),
+      )
+
+      evService.expectMsgType[DepartingEvsResponse] match {
+        case DepartingEvsResponse(evcs, evModels) =>
+          evcs shouldBe evcsInputModel.getUuid
+          evModels should have size 1
+          evModels.headOption match {
+            case Some(evModel) =>
+              evModel.uuid shouldBe ev1800.uuid
+              evModel.storedEnergy should approximate(KilowattHours(8.25))
+            case None => fail("Expected to get at least one ev.")
+          }
+      }
+
+      // TICK 5400: ev2700 leaves
+
+      evService.send(
+        evcsAgent,
+        DepartingEvsRequest(5400, Seq(ev2700.uuid)),
+      )
+
+      evService.expectMsgType[DepartingEvsResponse] match {
+        case DepartingEvsResponse(evcs, evModels) =>
+          evcs shouldBe evcsInputModel.getUuid
+          evModels should have size 1
+          evModels.headOption match {
+            case Some(evModel) =>
+              evModel.uuid shouldBe ev2700.uuid
+              evModel.storedEnergy should approximate(KilowattHours(16.5))
+            case None => fail("Expected to get at least one ev.")
+          }
+      }
+
+      // Check the results
+      // expect 7 Evcs Results and 12 Ev Results
+      // FixMe: 7 Evcs Results is correct but the one for tick 5400 is missing, the one at tick 0 comes twice...
+
+      Range(0, 19)
+        .map { _ =>
+          resultListener.expectMsgType[ParticipantResultEvent]
+        }
```

Comment:
I think it'd make sense to test for these events for the given times separately, because then we don't need `match` for the right times. So, after results for tick 900 should have been sent out, we're waiting for 4 results, etc.

Reply:
I find this one easier to understand because if we organise it the other way one always has to jump between the steps since the results for tick 900 will be available at tick 2700. But I'm not insisting on this.

Reply:
As discussed I tried to rearrange the results. But I'm not really happier with this solution. It produced ~100 lines more and is in my opinion still less understandable. (But in the end I would be able to accept also this one :) )

Reply:
I prepared a suggestion at two places where results are matched. Would've just done it a lot shorter like this. If the result type does not match, a `MatchError` is thrown, which is fine imo.

Reply:
Like the Evcs part very much and also took some of your improvements for Evs for the later test cases. Please have a look

Reply:
Thanks, looks great. With the new ParticipantAgent I intend to send results of the active tick always, so that'd be a bit more easy to grasp in the test.